src/foam/core/InvalidX.java

```diff
@@ -6,29 +6,43 @@
 
 package foam.core;
 
+import foam.nanos.logger.Logger;
+
 public class InvalidX
-  extends AbstractX
+  extends ProxyX
 {
-  private RuntimeException exception_;
+  private X delegate_;
+  private String nspecName_;
+  private Logger logger_;
 
-  public InvalidX(RuntimeException exception) {
-    exception_ = exception;
+  public InvalidX(X delegate, String nspecName, Logger logger) {
+    super(delegate);
+    delegate_ = delegate;
+    nspecName_ = nspecName;
+    logger_ = logger;
   }
 
+  @Override
   public Object get(X x, Object key) {
-    throw exception_;
+    Object ret = getX().get(x, key);
```

Comment:
Rather than copy this code from `ProxyX`, we should just change line 30 below from

```Java
  return ret;
```

to

```Java
  return super.get(x, key);
```